### ISSUE TITLE
[win10] jack conection

### ISSUE THREAD

**Issue by uidar** [aubio/aubio#205]
Jack is running with this command

> A:\VJ>"C:\Program Files (x86)\Jack\jackd" -R -S -d portaudio -d "ASIO::ASIO4ALL v2"

And now both

> A:\VJ\aubio-0.4.6-win64\bin> .\aubionotes.exe
> A:\VJ\aubio-0.4.6-win64\bin> .\aubionotes.exe -j

simply writes down help

>PS A:\VJ\aubio-0.4.6-win64\bin> .\aubionotes.exe -j x
AUBIO ERROR: onset: samplerate (0) can not be < 1
AUBIO ERROR: pitch: samplerate (0) can not be < 1

What should I type instead of x? Documentation says nothing...
I want to extract as much I can from mic input and direct midi output to somew VJing software.

**Comment by piem** [comment]
Hi @uidar 

Sorry, had never tried to run with jack on windows. How did you compile/install aubio?

The sampling rate should be fetched from the jack client in [utils.c](https://github.com/aubio/aubio/blob/master/examples/utils.c#L125). The default value of `0` can be changed in the [same file](https://github.com/aubio/aubio/blob/master/examples/utils.c#L40).

Did you try adding the option `-r`? Also adding the `-v` to see the full output.

    ./aubionotes.exe -j -r 44100 -v

If that doesn't help, could you try recompiling from the latest `HEAD` or `0.4.7`?

The source of the manual page is in [doc/aubionotes.txt](https://github.com/aubio/aubio/blob/master/doc/aubionotes.txt), if you see anything missing.

**Comment by uidar** [comment]
>A:\VJ\aubio-0.4.6-win64\bin> .\aubionotes.exe -r 44000 -v -j
Error: no arguments given (and no available audio input)
usage: A:\VJ\aubio-0.4.6-win64\bin\aubionotes.exe [ options ]
       -i      --input            input file
       -r      --samplerate       select samplerate
[ . . . ]

>A:\VJ\aubio-0.4.6-win64\bin> .\aubionotes.exe -r 44000 -v -j x
using source: x at 44000Hz
onset method: default, buffer_size: 512, hop_size: 256, threshold: 0.000000
pitch method: default, buffer_size: 2048, hop_size: 256, tolerance: 0.000000
usage: A:\VJ\aubio-0.4.6-win64\bin\aubionotes.exe [ options ]
       -i      --input            input file
       -r      --samplerate       select samplerate
[ . . . ]

I am using binaries 0.4.6 downloaded from 
>https://aubio.org/download

I am no sure how to compile latest ver under windows. I have no "make" tool. Should I install cygwin?

**Comment by uidar** [comment]
>A:\VJ\aubio-0.4.6-win64\bin> .\aubionotes.exe -j -r 44100 -v
Error: no arguments given (and no available audio input)
usage: A:\VJ\aubio-0.4.6-win64\bin\aubionotes.exe [ options ]
-i --input input file
-r --samplerate select samplerate
[ . . . ]

>A:\VJ\aubio-0.4.6-win64\bin> .\aubionotes.exe -j -r 44100 -v x
using source: x at 44100Hz
onset method: default, buffer_size: 512, hop_size: 256, threshold: 0.000000
pitch method: default, buffer_size: 2048, hop_size: 256, tolerance: 0.000000
usage: A:\VJ\aubio-0.4.6-win64\bin\aubionotes.exe [ options ]
-i --input input file
-r --samplerate select samplerate
[ . . . ]

I am using win binaries downloaded from

>https://aubio.org/download

I am no sure how to compile latest ver under windows. I have no "make" tool. Should I install cygwin?

**Comment by piem** [comment]
hi @uidar 

> another pile of problems

Not sure what else is missing. Note that only the command line tools in `examples` use JACK.

Do you have another program with jack support enabled on windows? Are you using the 32 or 64 bit version?

best, piem

**Comment by alfonso73** [comment]
hi @uidar 

how habe you been able to compile on MSYS2?
When i try i have this error


make[1]: Entering directory '/home/livemachine/aubio'
./scripts/get_waf.sh
/c/Users/Public/Documents/Wondershare/CreatorTemp/tmp.luSy35v18o /home/livemachine/aubio
gpg: Fatal: can't create directory '/c/Users/Public/Documents/Wondershare/CreatorTemp/tmp.luSy35v18o/C:\msys64\home\livemachine/.gnupg': No such file or directory
(23) Failed writing body
make[1]: *** [Makefile:53: getwaf] Error 2
make[1]: Leaving directory '/home/livemachine/aubio'
make: *** [Makefile:50: checkwaf] Error 2


thanks amd best

**Comment by piem** [comment]
hi @alfonso73

this is a different issue. it seems `gpg` is installed but fails creating a directory `~/.gnugp`. As a workaround, you could simply comment out line 36 of the script [`scripts/get_waf.sh`](/aubio/aubio/tree/master/scripts/get_waf.sh#L36).

Let us know if that helps; otherwise, please open a new issue.

cheers, piem

**Comment by alfonso73** [comment]
hi @piem . thanks for the info. Wich line should i exactly comment out? Line 36 is simply "checkwaf"

**Comment by piem** [comment]
yes @alfonso73, `checkwaf` is the function that checks the gpg signature of the downloaded waf.

you should be able to work around this issue by commenting this line.

**Comment by alfonso73** [comment]
@piem, i commented that line out. Building process started but it get stuck and i reported an issue here https://github.com/aubio/aubio/issues/256
